controllers/bookController.js

```diff
@@ -1,18 +1,56 @@
 const Book = require("../models/bookModel");
+const AppError = require("../utils/appError");
+const catchAsync = require("../utils/catchAsync");
 
-exports.addBook = async (req, res, next) => {
-  try {
-    const newBook = await Book.create(req.body);
-    res.status(201).json({
-      status: "Success",
-      data: {
-        book: newBook,
-      },
-    });
-  } catch (err) {
-    res.status(400).json({
-      status: "Fail",
-      errorMessage: err,
-    });
+exports.addBook = catchAsync(async (req, res, next) => {
```

Comment:
The book endpoints are not supposed to be part of this PR though

Reply:
yea, it was long before you told me authentication is the first thing to implement